components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/notification/mgt/NotificationManagementServiceImpl.java

```diff
@@ -158,6 +161,50 @@ public boolean updateNotificationStatus(int notificationId, Notification.Status
     }
 
     @Override
+    public PaginationResult getAllNotifications(PaginationRequest request) throws NotificationManagementException {
+        PaginationResult paginationResult = new PaginationResult();
+        List<Notification> notifications = new ArrayList<>();
+        int count =0;
+        try {
+            NotificationManagementDAOFactory.openConnection();
+            notifications = notificationDAO.getAllNotifications(request, NotificationDAOUtil.getTenantId());
+            count = notificationDAO.getNotificationCount(NotificationDAOUtil.getTenantId());
+            paginationResult.setData(notifications);
+            paginationResult.setRecordsFiltered(count);
+            paginationResult.setRecordsTotal(count);
+            return paginationResult;
+        } catch (SQLException e) {
+            throw new NotificationManagementException("Error occurred while opening a connection to" +
+                    " the data source", e);
+        } finally {
+            NotificationManagementDAOFactory.closeConnection();
+        }
+    }
+
+    @Override
+    public PaginationResult getNotificationsByStatus(Notification.Status status,
+                                                     PaginationRequest request) throws NotificationManagementException{
+        PaginationResult paginationResult = new PaginationResult();
+        List<Notification> notifications = new ArrayList<>();
+        int count =0;
+        try {
+            NotificationManagementDAOFactory.openConnection();
+            notifications = notificationDAO.getNotificationsByStatus(request, status, NotificationDAOUtil.getTenantId());
+            count = notificationDAO.getNotificationCountByStatus(status, NotificationDAOUtil.getTenantId());
```

Comment:
Do we need another method call for this? Can't we get the count from the size of 'notifications' list which we obtained from line#192.